services/iot/mock/mock.go

```diff
@@ -51,3 +51,21 @@ func (mock *client) GetUncompletedTasks(userID string) ([]api.TaskDescription, e
 	args := mock.Called(userID)
```

Comment:
Returnera `IoTClient` istället för `*client` i mock:en så kommer den vi få varningar då funktioner inte är implementerade i interfacet.

Reply:
do like this:
```
type client struct {
	mock.Mock
}

// Create returns an empty mock
func Create() *client {
	return new(client)
}

var _ iot.IoTClient = &client{}
```

Reply:
if you change from `Create() *client` to `Create() iot.IoTClient`, you won't be able to use the functions on `mock.Mock`